var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -52,7 +52,7 @@ class Llvm(Package):
     depends_on('cmake @2.8.12.2:')
 
     # Universal dependency
-    depends_on('python@2.7:')
+    depends_on('python@2.7:2.8')  # Seems not to support python 3.X.Y
```

Comment:
If you just want it to match 2.7.x, `depends_on('python@2.7')` should work.  If it doesn't it's a bug.